### PR TITLE
Support pretty option for _explain request

### PR DESCRIPTION
**Pull request by chloe-zh**
*Issue #, if available:*
* [#166 pretty option doesn't work for explain request](https://github.com/opendistro-for-elasticsearch/sql/issues/166)

*Description of changes:*
* Added a json pretty formatter, the formatter is called when  `_explain` with `?pretty` or `?pretty=true` is requested.
* Added IT for explain requests with pretty params.
* Added UT for the formatter.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/ExplainIT.java [hunk 1]
```diff
@@ -211,4 +211,37 @@ public void testContentTypeOfExplainRequestShouldBeJson() throws IOException {
 
         assertEquals("application/json; charset=UTF-8", response.getHeader("content-type"));
     }
+
+    @Test
+    public void assertExplainPrettyFormatted() throws IOException {
+        String query = String.format("SELECT firstname FROM %s", TestsConstants.TEST_INDEX_ACCOUNT);
```
Comment (davidcui-amzn) [inline src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/ExplainIT.java]: I think the general practice is to use our internal StringUtils.format() over String.format()

Comment (chloe-zh) [reply src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/ExplainIT.java]: Good point! Thanks!

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/ExplainIT.java [hunk 2]
```diff
@@ -211,4 +212,37 @@ public void testContentTypeOfExplainRequestShouldBeJson() throws IOException {
...
+        String query = StringUtils.format("SELECT firstname FROM %s", TEST_INDEX_ACCOUNT);
+
+        String notPrettyExplainOutputFilePath = TestUtils.getResourceFilePath(
+                "src/test/resources/expectedOutput/explainIT_format_not_pretty.json");
+        String notPrettyExplainOutput = Files.toString(new File(notPrettyExplainOutputFilePath), StandardCharsets.UTF_8);
+
+        assertThat(executeExplainRequest(query, ""), equalTo(notPrettyExplainOutput));
+        assertThat(executeExplainRequest(query, "pretty=false"), equalTo(notPrettyExplainOutput));
+
+        String prettyExplainOutputFilePath = TestUtils.getResourceFilePath(
```
Comment (galkk) [inline src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/ExplainIT.java]: Please extract that part to different test

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/JsonPrettyFormatter.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+
+import java.io.IOException;
+
+public class JsonPrettyFormatter {
+
+    public JsonPrettyFormatter() {
```
Comment (galkk) [inline src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/JsonPrettyFormatter.java]: This class doesn't carry any state. Consider making constructor private and make format static

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/ExplainIT.java (hunk 2 above)
Comment (chloe-zh) [reply src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/ExplainIT.java]: Done

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/JsonPrettyFormatter.java (hunk 3 above)
Comment (chloe-zh) [reply src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/JsonPrettyFormatter.java]: Good point thanks! It's done.

Comment (chloe-zh) [reply src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/JsonPrettyFormatter.java]: @galkk good point thanks! it's done.

**Pull request merged**